core/iwasm/fast-jit/fe/jit_emit_memory.c

```diff
@@ -408,9 +409,50 @@ jit_compile_op_memory_size(JitCompContext *cc)
     return false;
 }
 
+extern bool
+wasm_enlarge_memory(WASMModuleInstance *module, uint32 inc_page_count);
+
 bool
-jit_compile_op_memory_grow(JitCompContext *cc)
+jit_compile_op_memory_grow(JitCompContext *cc, uint32 mem_idx)
 {
+    JitReg delta, module, grow_result, res, memory_inst, prev_page_count;
+    JitInsn *insn;
+
+    /* WASMMemoryInstance->cur_page_count before enlarging */
+    memory_inst = get_memory_inst_reg(cc->jit_frame, mem_idx);
+    prev_page_count = jit_cc_new_reg_I32(cc);
+    GEN_INSN(LDI32, prev_page_count, memory_inst,
+             NEW_CONST(I32, offsetof(WASMMemoryInstance, cur_page_count)));
+
+    /* call wasm_enlarge_memory */
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    /* Set native_ret to x86::eax */
+    grow_result = jit_codegen_get_hreg_by_name("eax");
+#else
+    grow_result = jit_cc_new_reg_I32(cc);
+#endif
+    module = get_module_reg(cc->jit_frame);
+    POP_I32(delta);
+    insn = GEN_INSN(CALLNATIVE, grow_result,
+                    NEW_CONST(PTR, (uintptr_t)wasm_enlarge_memory), 2);
+    if (insn) {
+        *(jit_insn_opndv(insn, 2)) = module;
+        *(jit_insn_opndv(insn, 3)) = delta;
+    }
+
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    GEN_INSN(MOV, grow_result, grow_result);
+#endif
+
+    /* check if enlarge memory success */
+    res = jit_cc_new_reg_I32(cc);
+    GEN_INSN(CMP, cc->cmp_reg, grow_result, NEW_CONST(I32, 1));
```

Comment:
Had better check not zero, but not check whether it is 1, some systems' bool true might be not 1.
`CMP cmp_reg, grow_result, 0`, `SELECTNE`